src/triton_cli/parser.py

```diff
@@ -496,8 +465,97 @@ def parse_args(argv=None):
     parse_args_repo(subcommands)
     parse_args_server(subcommands)
     parse_args_inference(subcommands)
+    parse_args_optimize(subcommands)
     parse_args_profile(subcommands)
     parse_args_utils(subcommands)
     add_verbose_args([parser])
-    args = parser.parse_args(argv)
+
+    argv_ = argv if argv is not None else sys.argv[1:]
+    # If a passthrough command is passed as the first arg,
+    # special handling is needed.
+    need_special_handling = len(argv_) > 1 and argv_[0] in PASSTHROUGH_SUBCOMMANDS
+    if need_special_handling:
+        updated_args, pruned_args = prune_extra_subcommand_args(
+            argv_, subcommands.choices.keys()
+        )
+        args, unknown_args = parser.parse_known_args(updated_args)
+        args = add_unknown_args_to_args(args, pruned_args, unknown_args)
+    else:
+        args = parser.parse_args(argv)
     return args
+
+
+# ================================================
+# Helper functions
+# ================================================
+def build_command(executable: str, args: argparse.Namespace):
+    skip_args = ["func", "task"]
+    cmd = [executable]
+    program_subcommand = None
+    for arg, value in vars(args).items():
+        if arg in skip_args:
+            pass
+        elif arg in PASSTHROUGH_SUBCOMMANDS:
+            program_subcommand = arg
+        elif value is False:
+            pass
+        elif value is True:
+            if len(arg) == 1:
+                cmd += [f"-{arg}"]
+            else:
+                cmd += [f"--{arg}"]
```

Comment:
Is this how we do this in the rest of our Python code? I generally like if/else statements because it's clear what's happening (the other code is more concise, but slower and less clear to read IMO). I'm happy to follow whatever convention we generally use though for consistency.

Reply:
It's definitely an optional and stylistic choice. I, personally, just find it more readable when the logic within each block is a a initialization-like one-liner. But I'm happy to leave it as-is as well!

Reply:
I'm super open. I want to be consistent with the rest of our codebase. If we don't have a general rule on this yet, I'll leave it as is if that's okay. I'm finding documentation online that the ternary operator is best reserved for simple cases, and this is less simple for me then something like `min = a if a < b else b`.

Reply:
I don't believe we have a general rule for this project, so let's just leave as-is.